Sources/_InternalTestSupport/Observability.swift

```diff
@@ -103,6 +103,18 @@ public func XCTAssertNoDiagnostics(
     XCTFail("Found unexpected diagnostics: \n\(description)", file: file, line: line)
 }
 
+public func expectNoDiagnostics(
+    _ diagnostics: [Basics.Diagnostic],
+    problemsOnly: Bool = true,
+    sourceLocation: SourceLocation = #_sourceLocation
+) {
+    let diagnostics = problemsOnly ? diagnostics.filter { $0.severity >= .warning } : diagnostics
+    if diagnostics.isEmpty { return }
+
+    let description = diagnostics.map { "- " + $0.description }.joined(separator: "\n")
+    Issue.record("Found unexpected diagnostics: \n\(description) \(sourceLocation.fileName):\(sourceLocation.line)")
```

Comment:
issue (if-patch-updated): instead of putting the source location in the error message, can we call `Issue.record` with the `sourceLocation` argument set.  That's, Swift Testing will accurately report the filename and line number.

```suggestion
    Issue.record("Found unexpected diagnostics: \n\(description)", sourceLocation: sourceLocation)
```

See https://developer.apple.com/documentation/testing/issue/record(_:sourcelocation:)